### PR TITLE
Shaded Hills fixes for August 20, 2024

### LINKED ISSUE #61: default link text on dark mode is unreadable

### LINKED ISSUE #62: can't clean hands with water
walking into water doesn't clean your hands, and you can't click on water to do it. nor does rain

### LINKED ISSUE #63: possums need butchery hook offsets

### LINKED ISSUE #64: mice/deer flee while lying down without updating icon

### LINKED ISSUE #65: border posts description should mention usage
border posts should say what they do in their description

### LINKED ISSUE #66: drake self-healing appears to be broken
todo: more info

### LINKED ISSUE #67: butchery does not return arrows embedded in mobs
simplemob butchery should probably clear/drop implants, which i think are qdeleted by default

### LINKED ISSUE #72: slicing cleaned large offal makes it need to be cleaned again

### LINKED ISSUE #75: Opossums immediately get back up after playing dead

### LINKED ISSUE #4384: Certain simple animals like cows speak as Unknown
a recent change to make GetVoice use real_name instead of name meant any mob without real_name set is named Unknown :(

### PR DESCRIPTION
**Pull request by out-of-phaze**
## Description of changes
Some fixes, edits, etc. The commit names are pretty descriptive. Some notes:
- Not all of the posture/AI changes are linked directly with existing issues, I found them while looking into the opossum stuff. would like a closer look at it, there were a few bugs in there that aren't on the tracker so i'm happy to explain
- I put the signpost info on mechanics info in the codex rather than on examine.

## Why and what will this PR improve
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/66
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/64
Fixes mice always appearing to be resting after an icon update, unless they were non-deliberately prone.
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/75
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/67
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/65
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/63
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/62
Fixes #4384
Improves saycode quality by removing a parameter that was never set by anything ever. Also adds `\The` so you get `The cow moos, "Moo."` instead of `cow moos, "Moo."`.
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/72.
Makes thread have less matter, so that it cleanly divides the matter amount in offal.
Fixes https://github.com/PyrelightSS13/Pyrelight/issues/61

## Authorship
Me. My girlfriend helped with the colors.

### CODE REVIEW

File: code/datums/ai/_ai.dm [hunk 1]
```diff
@@ -141,18 +141,19 @@
...
-	else if(!stop_wander && !body.buckled_mob && do_wander && !body.anchored)
-		if(isturf(body.loc) && !body.current_posture?.prone)		//This is so it only moves if it's not inside a closet, gentics machine, etc.
-			turns_since_wander++
-			if(turns_since_wander >= turns_per_wander && (!(stop_wander_when_pulled) || !LAZYLEN(body.grabbed_by))) //Some animals don't move when pulled
-				var/direction = pick(wander_directions || global.cardinal)
-				var/turf/move_to = get_step(body.loc, direction)
-				if(body.turf_is_safe(move_to))
-					body.SelfMove(direction)
-					turns_since_wander = 0
+	else if(isturf(body.loc))		//This is so it only moves if it's not inside a closet, gentics machine, etc.
```
Comment (MistakeNot4892) [inline code/datums/ai/_ai.dm:149]: This -should- be handled by STANCE_CONTAINED currently.

File: code/modules/butchery/butchery_products.dm [hunk 2]
```diff
@@ -133,6 +133,7 @@
 	for(var/obj/item/food/butchery/offal/guts in .)
 		if(dry && !guts.dry)
 			guts.dry = TRUE
+			guts._cleaned = TRUE // so that it doesn't prompt you to re-cut it; dried offal is always cleaned
```
Comment (MistakeNot4892) [inline code/modules/butchery/butchery_products.dm]: This isn't a correct fix. I think you just need to remove `else` from the block immediately below this one.

File: code/modules/butchery/butchery_products.dm [hunk 3]
```diff
@@ -153,6 +154,9 @@
 		var/obj/item/food/dried = new dried_type(loc, (material.tans_to || material.type))
 		if(istype(dried))
 			dried.dry = TRUE
+			if(istype(dried, /obj/item/food/butchery/offal))
```
Comment (MistakeNot4892) [inline code/modules/butchery/butchery_products.dm]: Drying isn't supposed to set cleaned universally, this is a bit of a confusing change

File: code/modules/mob/living/simple_animal/_simple_animal.dm [hunk 4]
```diff
@@ -157,7 +157,7 @@ var/global/list/simplemob_icon_bitflag_cache = list()
 		icon_state += "-dead"
 	else if(stat == UNCONSCIOUS && (mob_icon_state_flags & MOB_ICON_HAS_SLEEP_STATE))
 		icon_state += "-sleeping"
-	else if(current_posture?.deliberate && (mob_icon_state_flags & MOB_ICON_HAS_REST_STATE))
+	else if(current_posture?.prone && (mob_icon_state_flags & MOB_ICON_HAS_REST_STATE))
```
Comment (MistakeNot4892) [inline code/modules/mob/living/simple_animal/_simple_animal.dm:160]: This is fixed in the fox/deer PR I think. Identical fix so shouldn't conflict at least.

File: code/datums/ai/_ai.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/datums/ai/_ai.dm:149]: this code was already there, i just de-indented it

File: code/modules/butchery/butchery_products.dm (hunk 2 above)
Comment (out-of-phaze) [reply code/modules/butchery/butchery_products.dm]: nope, just actually needed to make the `!_cleaned` check also check `!dry`. which is identical to this change in functionality really, just cleaner. as you said, cleaned and dry aren't meant to coexist

File: code/datums/ai/_ai.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/datums/ai/_ai.dm:149]: yeah STANCE_CONTAINED and other stances are only used by aggressive mobs, not the base AI or passive mobs etc

File: mods/content/fantasy/props/signpost.dm [hunk 5]
```diff
@@ -53,6 +53,10 @@
 	despawn_character(user)
 	return TRUE
 
+/obj/effect/departure_signpost/get_mechanics_info()
+	. = ..()
+	. += "By clicking on the signpost, you can choose to remove your character from the round in order to free your job slot and respawn. This will delete your character's worn and held items, so make sure not to take any important items with you."
```
Comment (MistakeNot4892) [inline mods/content/fantasy/props/signpost.dm:58]: Do we want a span class to make this more visible?

**Review by MistakeNot4892 — APPROVED** [review]

File: mods/content/fantasy/props/signpost.dm (hunk 5 above)
Comment (out-of-phaze) [reply mods/content/fantasy/props/signpost.dm:58]: it's in the codex

Comment (MistakeNot4892) [reply mods/content/fantasy/props/signpost.dm:58]: Per Discord I really think this should be in examine or something, I don't think anyone is going to see it in the codex link.

**Pull request merged**